qcodes/instrument_drivers/QDev/QDac_channels.py

```diff
@@ -0,0 +1,684 @@
+        self._write_response = ''
+
+        # The following bool is used in self.write
+        self.debugmode = False
+
+        if self._get_firmware_version() < 0.170202:
+            raise RuntimeError('''
+                               Obsolete QDAC Software version detected.
+                               QCoDeS only supports version 0.170202 or newer.
+                               Contact rikke.lutge@nbi.ku.dk for an update.
+                               ''')
+
+        self.num_chans = num_chans
+
+        # Assigned slopes. Entries will eventually be [chan, slope]
+        self._slopes = []
+        # Function generators (used in _set_voltage)
+        self._fgs = set(range(1, 9))
+        self._assigned_fgs = {}  # {chan: fg}
+        # Sync channels
+        self._syncoutputs = []  # Entries: [chan, syncchannel]
+
+        self.chan_range = range(1, 1 + self.num_chans)
+        self.channel_validator = vals.Ints(1, self.num_chans)
+
+        channels = ChannelList(self, "Channels", QDacChannel,
+                               snapshotable=False,
+                               paramclass=QDacMultiChannelParameter)
+
+        for i in self.chan_range:
+            channel = QDacChannel(self, 'chan{}'.format(i), i)
+            channels.append(channel)
+            # Should raise valueerror if name is invalid (silently fails now)
+            self.add_submodule('ch{:02}'.format(i), channel)
+        channels.lock()
+        self.add_submodule('channels', channels)
+
+        for board in range(6):
+            for sensor in range(3):
+                label = 'Board {}, Temperature {}'.format(board, sensor)
+                self.add_parameter(name='temp{}_{}'.format(board, sensor),
+                                   label=label,
+                                   unit='C',
+                                   get_cmd='tem {} {}'.format(board, sensor),
+                                   get_parser=self._num_verbose)
+
+        self.add_parameter(name='cal',
+                           set_cmd='cal {}',
+                           vals=self.channel_validator)
+        # TO-DO: maybe it's too dangerous to have this settable.
+        # And perhaps ON is a better verbose mode default?
+        self.add_parameter(name='verbose',
+                           set_cmd='ver {}',
+                           val_mapping={True: 1, False: 0})
+
+        # Initialise the instrument, all channels DC (unbind func. generators)
+        for chan in self.chan_range:
+            # Note: this call does NOT change the voltage on the channel
+            self.write('wav {} 0 1 0'.format(chan))
+
+        self.verbose.set(False)
+        self.connect_message()
+        log.info('[*] Querying all channels for voltages and currents...')
+        self._get_status(readcurrents=update_currents)
+        log.info('[+] Done')
+
+    #########################
+    # Channel gets/sets
+    #########################
```

Comment:
I think it makes sense for this particular instrument to keep it in the main instrument, as the channels are interlinked via the `_get_status` command.